### PR TITLE
Add priorityPreemptionPolicyValue as a configurable value on the datadog

### PR DESCRIPTION
**Pull request by mrvarmazyar**
#### What this PR does / why we need it:
On the GKE spot instances, I needed to configure the `preemptionPolicy` to Never, which wasn't available as a configurable value on the helm charts. 
In this PR, I added the `priorityPreemptionPolicyValue` as a configurable value on the agent charts 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Datadog changelog
 
+## 2.36.7
+* Add `priorityPreemptionPolicyValue` as a configurable value on the agent charts
```
Comment (urseberry) [inline charts/datadog/CHANGELOG.md:4]:
```suggestion
* Add `priorityPreemptionPolicyValue` as a configurable value on the Agent charts
```

File: charts/datadog/README.md [hunk 2]
```diff
@@ -518,6 +518,7 @@ helm install --name <RELEASE_NAME> \
 | agents.priorityClassCreate | bool | `false` | Creates a priorityClass for the Datadog Agent's Daemonset pods. |
 | agents.priorityClassName | string | `nil` | Sets PriorityClassName if defined |
 | agents.priorityClassValue | int | `1000000000` | Value used to specify the priority of the scheduling of Datadog Agent's Daemonset pods. |
+| agents.priorityPreemptionPolicyValue | string | `"PreemptLowerPriority"` | It can also be changed to Never in order to change the PriorityClass to non-preempting  |
```
Comment (urseberry) [inline charts/datadog/README.md:521]:
```suggestion
| agents.priorityPreemptionPolicyValue | string | `"PreemptLowerPriority"` | Set to Never to change the PriorityClass to non-preempting  |
```

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -1369,6 +1369,9 @@ agents:
   # agents.priorityClassName -- Sets PriorityClassName if defined
   priorityClassName:
 
+  # agents.priorityPreemptionPolicyValue -- It can also be changed to Never in order to change the PriorityClass to non-preempting
```
Comment (urseberry) [inline charts/datadog/values.yaml:1372]:
```suggestion
  # agents.priorityPreemptionPolicyValue -- Set to "Never" to change the PriorityClass to non-preempting 
```

**Review by urseberry — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]
Thanks @mrvarmazyar for this contribution

Your change looks good. I will validate the PR and merge it when the will be green 👍

**Comment by mrvarmazyar** [comment]
@clamoriniere You're welcome; it's my pleasure!
Should I fix the lint-docs, or it's gonna be fixed as part of the validation process?

**Comment by clamoriniere** [comment]
> Should I fix the lint-docs, or it's gonna be fixed as part of the validation process?

Yes please fix the same linter issue

**Pull request merged**